orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/pipeline/DeleteIntentStage.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.keel.pipeline
+
+import com.netflix.spinnaker.orca.Task
+import com.netflix.spinnaker.orca.keel.task.DeleteIntentTask
+import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder
+import com.netflix.spinnaker.orca.pipeline.TaskNode
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.springframework.stereotype.Component
+import kotlin.reflect.KClass
+
+@Component
+class DeleteIntentStage() : StageDefinitionBuilder {
+  override fun taskGraph(stage: Stage, builder: TaskNode.Builder) {
+    builder.withTask("deleteIntent", DeleteIntentTask::class)
+  }
+
+  private fun TaskNode.Builder.withTask(name: String, type: KClass<out Task>) =
+    withTask(name, type.java)
+}
```

Comment:
A common pattern for that kind of thing would be to declare an extension function:

```kotlin
inline fun <reified T: Class<Task>> StageDefinitionBuilder.withTask(name: String) =
  withTask(name, T::class.java)
```

Which you could then call with `builder.withTask<DeleteIntentTask>("deleteIntent")`.

Reply:
I copied the style from https://github.com/spinnaker/orca/blob/master/orca-dry-run/src/main/kotlin/com/netflix/spinnaker/orca/dryrun/DryRunStage.kt#L39

I'm gonna ask you to explain this to me offline